### ISSUE TITLE
OPCOST: Update script to install RODBC if not present

### ISSUE THREAD

**Issue by lbross** [USFS-PNW/Fia-Biosum-Manager#115]
If you try to run OpCost and haven't installed the RODBC package, you receive a cryptic error message. Per @sloreno the following change will install the package if the user doesn't already have it:

require(RODBC) || install.packages(RODBC)
replaces:
library(RODBC)

**Comment by lbross** [comment]
I have made the update to the script. Leaving the OpCost version (and file name) the same because this is a minor change and doesn't warrant the additional work required to change the version number

**Comment by lbross** [comment]
This directive didn't work for @lpotts. He had to install RODBC manually. I don't know enough about R to pursue further.

**Comment by lbross** [comment]
I was able to remove RODBC from my computer by first location the library paths using this command: .libPaths(). Then deleted RODBC folders from any of these paths. @sloreno provided a package.check directive that did the trick. Closing.

**Comment by lbross** [comment]
A couple of caveats on this from my experience on Windows 10.
1. In order for the script to install RODBC you may need to run FIA Biosum as an administrator
2. You must select Rscript.exe in the i386 directory. I picked the one in the bin directory the first time and it didn't work.

If a user ever has trouble with opcost_input not being created, always start by looking at opcost_error_log.txt in the AppData\Local\Temp directory.

**Comment by carlinstarrs** [comment]
I uninstalled RODBC by using remove.packages(RODBC) and then tried to run processor again. I got the following error: 

![image](https://user-images.githubusercontent.com/12533482/37008212-eff0ca32-2095-11e8-8912-935f6da7341d.png)

I tried again running it as administrator and it gave me the same error. I did have the 32-bit selected (i386 directory): 
![image](https://user-images.githubusercontent.com/12533482/37008387-9eca7dfa-2096-11e8-9ca9-9afc7cc0b777.png)

The opcost_error_log says:
Error in library(RODBC) : there is no package called 'RODBC'
Execution halted

If I run it from the R script directly (9_1.R) it does work. It does throw an error saying there is no package called 'RODBC', but R shows the package as installed and active. It may be that the OPCOST script is seeing that error and stopping even though it doesn't need to (I think it's a weird quirk of the library() function).

You might try replacing lines 3-10 in 9_1.R with: 

```
is.installed <- function(mypkg){
  is.element(mypkg, installed.packages()[,1])
} 

if (!is.installed("RODBC")){
  install.packages("RODBC")
}

library("RODBC")
```

I would but it's saying I don't have adequate permissions and it won't let me change it for some reason. This script will be slower but it does not throw any false errors.

**Comment by lbross** [comment]
This is a tough one because it seems to work differently depending on the computer. What editor are you using to edit the R script? Can you try starting that up as an Administrator and see if it lets you make the recommended changes? I can change it and send it to you but if you can't edit the script, you likely can't replace the file either.

**Comment by carlinstarrs** [comment]
I'm using RStudio, and it will let me edit the file but not save my edits, and I can't replace the file in explorer either.

**Comment by lpotts** [comment]
What directory are you trying to edit and save the file? -Larry

**Comment by carlinstarrs** [comment]
C:\Program Files (x86)\FIA PNW Portland Forestry Sciences Lab\FIA Biosum 5.8.0\opcost

**Comment by jsfried** [comment]
Larry-- it works better to post directly on Github, to avoid the repetition of material from earlier in the thread. Carlin, there is no reason to keep OpCost in the directory where it installs. I always keep it on my external drive where the projects are located and can edit there at will. Since you point BioSum to the location, it can be anywhere.

**Comment by carlinstarrs** [comment]
Alright, so this seems to fix it. You should not need to run as administrator. It does slow it down a bit but
not too much. 

```
is.installed <- function(mypkg){
  is.element(mypkg, installed.packages()[,1])
} 

if (!is.installed("RODBC")){
  install.packages("RODBC", repos = "http://cran.us.r-project.org")
}

library("RODBC")
```

**Comment by carlinstarrs** [comment]
Oh, and FYI if you use remove.packages("RODBC") to test, it seems you need to close/restart RStudio (and maybe whatever else you would be using) for it to fully remove

**Comment by lbross** [comment]
Thanks Carlin. We are waiting for a new OpCost script from Conor with some other fixes so I will pre-pend your lines to that script when I get it. Hopefully you can test it one last time to make sure I got it right? I will send you the new script when I have it. Reopening until Carlin has verified the update on the forthcoming new script.

**Comment by lbross** [comment]
Resolved with OpCost 10.0!
